api/include/vehicles/kia_soul_niro.h

```diff
@@ -0,0 +1,420 @@
+/**
+ * @file kia_soul_niro.h
+ * @brief Kia Soul Niro specific macros.
+ *
+ */
+
+
+#ifndef _KIA_SOUL_NIRO_PLATFORM_INFO_H_
+#define _KIA_SOUL_NIRO_PLATFORM_INFO_H_
+
+
+#include <stdint.h>
+
+
+// ********************************************************************
+//
+// WARNING
+//
+// The values listed here are carefully tested to ensure that the vehicle's
+// components are not actuated outside of the range of what they can handle.
+// By changing any of these values you risk attempting to actuate outside of the
+// vehicle's valid range. This can cause damage to the hardware and/or a
+// vehicle fault. Clearing this fault state requires additional tools.
+//
+// It is NOT recommended to modify any of these values without expert knowledge.
+//
+// ************************************************************************
+
+
+// ****************************************************************************
+// OBD MESSAGES
+// ****************************************************************************
+
+/*
+ * @brief ID of the Kia Soul's OBD steering wheel angle CAN frame.
+ *
+ */
+#define KIA_SOUL_OBD_STEERING_WHEEL_ANGLE_CAN_ID ( 0x2B0 )
+
+/*
+ * @brief ID of the Kia Soul's OBD wheel speed CAN frame.
+ *
+ */
+#define KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID ( 0x4B0 )
+
+/*
+ * @brief ID of the Kia Soul's OBD brake pressure CAN frame.
+ *
+ */
+#define KIA_SOUL_OBD_BRAKE_PRESSURE_CAN_ID ( 0x220 )
+
+/*
+ * @brief Factor to scale OBD steering angle to degrees
+ *
+ */
+#define KIA_SOUL_OBD_STEERING_ANGLE_SCALAR ( 0.1 )
+
+/**
+ * @brief Steering wheel angle message data.
+ *
+ */
+typedef struct
+{
+    int16_t steering_wheel_angle; /* 1/10 degrees */
+
+    uint8_t reserved[6]; /* Reserved. */
+} kia_soul_obd_steering_wheel_angle_data_s;
+
+/**
+ * @brief Wheel speed message data.
+ *
+ */
+typedef struct
+{
+    int16_t wheel_speed_front_left; /* 1/50 mph */
```

Comment:
Wheel speeds are only the lower byte of each so they're bytes 0, 2, 4, 6 as uint8_t and seem to be around 1/4 for kph

Reply:
9394cc0